src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaAttribute.cs

```diff
@@ -139,5 +148,17 @@ public KafkaAttribute()
         /// ssl.key.password in librdkafka
         /// </summary>
         public string SslKeyPassword { get; set; }
+
+        /// <summary>
```

Comment:
Properties should have getters, according [.NET guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/property).

I can imagine that we might want to log/write the value and not having the setters prevents it.

Reply:
In my latest commit I have used one possible implementation. The getters have been reintroduced, in order to comply with guidelines, but the backing fields are still nullable. The values from the attribute are then merged (coalesced) with the values from the global options (can be also done as a one-liner using JsonConvert, among many ways of doing it), and the default value is used if neither supplies it. 

The same approach could definitely be used for consumers as well. It may seem a bit convoluted, and there may well be better ways of doing it, but I think it provides maximum flexibility, while maintaining backwards compatibility.

Reply:
The merging could also look like this:

```
var mergedOptions = DefaultProducerOptions.CreateNew()
                    .MergeWithNonNullPropertiesFrom(kafkaOptions)
                    .MergeWithNonNullPropertiesFrom(entity.Attribute);
```

using a simple extension like this:

```
public static class MergeExtensions
{
    public static T MergeWithNonNullPropertiesFrom<T>(this T instance, T other)
    {
        if (other != null)
        {
            JsonConvert.PopulateObject(JsonConvert.SerializeObject(other, new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore }), instance);
        }
        return instance;
    }
}
```